static/src/javascripts/projects/common/utils/detect.js

```diff
@@ -71,6 +67,72 @@ define([
         ],
         detect;
 
+    var breakpointNames = breakpoints.map(getBreakpointName);
+
+    var currentBreakpoint;
+    var currentTweakpoint;
+
+    init(window);
+
+    function init(win) {
+        if ('matchMedia' in win) {
+            initMediaQueryListeners(win);
+        } else {
+            updateBreakpoints.call(win);
+            win.addEventListener('resize', debounce(updateBreakpoints, 200));
```

Comment:
yeah it wasn't the animationframe so much as a single listener throttling once, rather than multiple listeners all throttling. i went with animation frame in the scroll listener to try and always prioritise scrolling rather than listening

> The developer needs to apply judgment as to what to do in a listener vs in an animation frame, but it should remain an informed choice

i'm not sure how i feel about this. creating that throttled event is a way of saying 'we prioritise perf over anything - you cannot have a better reason to do something, so this is all you need'. if there is a case where we *need* to block rendering the original scroll event is still there. but isn't it a fairly common pattern to abstract best practise into easily consumable apis? especially when the code has a lot of contributors who may only touch it occasionally with specific features or goals in mind. the intention was to create a frictionless way of implementing best-practise derived from a holistic, rather than case-specific, viewpoint